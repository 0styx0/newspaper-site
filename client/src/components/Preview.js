import React from 'react';
import { Link } from 'react-router-dom';

function Preview(props) {

    return (
        <div className="preview">
            <div className="content" dangerouslySetInnerHTML={{__html: props.lede}} />

            <Link
              className="small"
              to={`/issue/${props.issue}/story/${props.url}`}
            >
              Read More
            </Link>
            <span className="small"> ({props.views} views)</span>
        </div>
    )
}



export default Preview;