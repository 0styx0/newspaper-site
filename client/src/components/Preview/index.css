.preview {
    text-shadow: none;
    padding: 7px;
    margin: 20px;
    border: 1px solid black;
    position: relative;
    top: 30px;
    width: 220px;
    height: 300px;
    text-align: center;
}


.preview h1, .preview h2, .preview h3 {
    font-size: 0.9em;
    text-shadow: 0px 0px 0px #000;
    margin: 0;
    margin-top: 15px;
    padding: 0;
}

 .preview h4, .preview h5, .preview h6 {
    font-size: 0.7em;
 }

.preview h1, .preview h2, .preview h3, .preview h4, .preview h5, .preview h6 {
    margin-bottom: 5px;
}

.preview p {
    margin-top: 19px;
    text-indent: 1.2em;
}

.preview  a {
    position: relative;
    top: 4px;
}

.preview span {
    position: relative;
    top: 4px;

}

.previewContent {
    overflow: hidden;
    text-align: left;
    height: 270px;
}

/* just in case*/
.content img {
    display: none;
}

/* creates fade */
.previewContent:after {
  position: absolute;
  bottom: 37px;
  height: 40%;
  width: 96%;
  content: "";
  background: linear-gradient(to top,
     rgba(253,247,239, 1),
     rgba(253,247,239, 0)
  );

}

.previewContent {
    margin-top: 10px
}

.previewHidden {
    display: none;
}

