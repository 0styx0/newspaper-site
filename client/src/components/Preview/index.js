import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './index.css';

export default function Preview(props) {

    return (
        <div className="preview">
            <div className="content" dangerouslySetInnerHTML={{__html: props.lede}} />

            <Link
              className="small"
              to={`/issue/${props.issue}/story/${props.url}`}
            >
              Read More
            </Link>
            <span className="small"> ({props.views} views)</span>
        </div>
    )
}

Preview.propTypes = {
    lede: PropTypes.string.isRequired,
    views: PropTypes.number.isRequired,
    issue: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
}