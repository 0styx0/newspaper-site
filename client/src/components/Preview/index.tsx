import * as React from 'react';
import { Link } from 'react-router-dom';

import './index.css';

interface Props {
    lede: string;
    views: number;
    issue: number;
    url: string;
}

export default function Preview(props: Props) {

    return (
        <div className="preview">
            <div className="content" dangerouslySetInnerHTML={{__html: props.lede}} />

            <Link
              className="small"
              to={`/issue/${props.issue}/story/${props.url}`}
            >
              Read More
            </Link>
            <span className="small"> ({props.views} views)</span>
        </div>
    )
}
