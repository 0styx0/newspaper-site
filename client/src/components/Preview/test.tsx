import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import Preview from './';

describe('<Preview>', () => {

    describe('snapshots', () => {

        it('should render correctly', () => {

            const tree = renderer.create(
                <MemoryRouter>
                    <Preview
                        url="random_string"
                        issue={21}
                        lede="<p>This is a lede</p>"
                        views={34}
                    />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
