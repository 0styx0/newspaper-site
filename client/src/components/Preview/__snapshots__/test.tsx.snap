// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview> snapshots should render correctly 1`] = `
<div
  className="preview"
>
  <div
    className="previewContent"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>This is a lede</p>",
      }
    }
  />
  <a
    className="small"
    href="/issue/21/story/random_string"
    onClick={[Function]}
  >
    Read More
  </a>
  <span
    className="small"
  >
     (
    34
     views)
  </span>
</div>
`;
